lib/api/CDataSummarizationJsonSerializer.cc

```diff
@@ -71,14 +71,14 @@ bool getBool(const rapidjson::Value& element, const std::string& tag, bool fallb
     return element[tag].GetBool();
 }
 
-std::stringstream decompressStream(boost::iostreams::stream_buffer<Device>& buffer) {
-    std::stringstream decompressedStream;
+auto decompressStream(boost::iostreams::stream_buffer<Device>& buffer) {
+    auto decompressedStream = std::make_shared<std::stringstream>();
```

Comment:
Is this change really necessary?  The returned stringstream will be moved, so returning by value won't be copying the contents.

Reply:
Ultimately, I don't think we want to be copying this into a `stringstream` at all. In fact, I started to make this change, which is relatively easy, but then thought, probably we should actually be reusing `CStateDecompressor` here and using a handler rather than parsing everything into intermediate representation.

I decided not to focus on that in this change, but didn't revert this small change because it represents a step in that direction.

Reply:
Yes, agreed it would be nice to reuse the pure streaming implementation we have rather than hold intermediate states in full.